src/webauth/agent.js

```diff
@@ -1,12 +1,12 @@
-import { NativeModules, Linking } from 'react-native';
+import {NativeModules, Linking, Platform} from 'react-native';
 
 export default class Agent {
-  show(url, closeOnLoad = false) {
+  show(url, ephemeralSession = false, closeOnLoad = false) {
```

Comment:
This agent class is public and I think a change like the one introduced here should be considered breaking. 

You are adding a parameter in the middle, those calling today with `show("URL", true) would have a different outcome. Please move it to the end and keep the defaults.

If instead, you consider this class as internal, let's document that and use a "configuration object" instead to make it future proof.

Reply:
I consider that class to be internal. How would a configuration object future-proof an internal class?

Reply:
My point is that if we keep adding parameters this signature won't look nice. It's OK because it's internal 👍 

A future-proof approach, that doesn't require changing the signature, would be:

```js
show(url, options = {})
const closeOnLoad = options.closeOnLoad || false
const ephemerealSession = options.ephemerealSession || false
```

Reply:
Shouldn't we remove `closeOnLoad`, anyway? Seems like old leftover code that doesn't do anything now. The native iOS code receives it as a parameter but does nothing with it. Same with the Java code.

Reply:
iOS uses natively [here](https://github.com/auth0/react-native-auth0/blob/master/ios/A0Auth0.m#L216-L220). 

A side effect of the parameter is to know when to return a value. It's checked [here](https://github.com/auth0/react-native-auth0/blob/master/src/webauth/agent.js#L24-L28). From what I can tell, that's only in use for [authentication](https://github.com/auth0/react-native-auth0/blob/master/src/webauth/index.js#L72) and for [logout](https://github.com/auth0/react-native-auth0/blob/master/src/webauth/index.js#L131).

Reply:
Good idea, but ATM it is just 3 parameters, which is fine IMO. If it keeps growing converting it to an options object is trivial.